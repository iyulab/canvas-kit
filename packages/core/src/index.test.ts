import { CanvasKitRenderer, Scene } from './index';
import { describe, it, expect, beforeEach, vi } from 'vitest';

describe('CanvasKitRenderer', () => {
  let renderer: CanvasKitRenderer;
  let canvas: HTMLCanvasElement;
  let scene: Scene;

  beforeEach(() => {
    canvas = document.createElement('canvas');
    renderer = new CanvasKitRenderer(canvas);
    scene = new Scene();
  });

  it('should clear the canvas', () => {
    const ctx = canvas.getContext('2d');
    if (!ctx) return;
    const clearRectSpy = vi.spyOn(ctx, 'clearRect');
    renderer.clear();
    expect(clearRectSpy).toHaveBeenCalledWith(0, 0, canvas.width, canvas.height);
  });

  it('should render a scene with a rectangle and a circle', () => {
    scene.add({ type: 'rect', x: 10, y: 20, width: 50, height: 100, fill: 'red' });
    scene.add({ type: 'circle', x: 30, y: 40, radius: 25, fill: 'blue' });

    renderer.render(scene);

    const ctx = canvas.getContext('2d');
    if (!ctx) return;
    // @ts-ignore
    expect(ctx.__getDrawCalls()).toMatchSnapshot();
  });
});
