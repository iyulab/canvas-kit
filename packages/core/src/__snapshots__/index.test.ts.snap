// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CanvasKitRenderer > should render a scene with a rectangle and a circle 1`] = `
[
  {
    "props": {
      "height": 150,
      "width": 300,
      "x": 0,
      "y": 0,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "clearRect",
  },
  {
    "props": {
      "height": 100,
      "width": 50,
      "x": 10,
      "y": 20,
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillRect",
  },
  {
    "props": {
      "fillRule": "nonzero",
      "path": [
        {
          "props": {},
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        {
          "props": {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 25,
            "startAngle": 0,
            "x": 30,
            "y": 40,
          },
          "transform": [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
]
`;
